Lib/fractions.py

```diff
@@ -35,6 +35,17 @@ def _gcd(a, b):
         a, b = b, a%b
     return a
 
+
+def _flat_divmod(na, da, nb, db):
+    """(a // b, a % b)"""
+    # div = a // b
+    n_div, d_div = na * db, da * nb
+    div = n_div // d_div
+    # mod = a - b * div == (na*db - da*nb * div) / (da*db)
```

Comment:
why this comments?

Reply:
The comments are there to explain what the code does mathematically. I don't think it's obvious from the operations otherwise.